### PR TITLE
Dev/issue 10581 exclude type fixes/cleanup

### PR DESCRIPTION
**Pull request by mcantelon**
* Removed un-implemented task option
* Added task option to display available document types
* Added mention of new task option to task's help text
* Fixed inaccurate comment in search plugin code

### CODE REVIEW

File: lib/task/search/arPopulateTask.class.php [hunk 1]
```diff
@@ -44,17 +44,33 @@ protected function configure()
...
     $excludeTypes = (!empty($options['exclude-types'])) ? explode(',', strtolower($options['exclude-types'])) : null;
 
     QubitSearch::getInstance()->populate($excludeTypes);
   }
+
+  protected function availableDocumentTypes()
+  {
+    $types = array_keys(QubitSearch::getInstance()->loadMappings()->asArray());
+    sort($types);
+    return $types;
```
Comment (MikeFE) [inline lib/task/search/arPopulateTask.class.php]: return sort($types); ?

File: lib/task/search/arPopulateTask.class.php [hunk 2]
```diff
@@ -44,17 +44,33 @@ protected function configure()
...
+    }
+
     new sfDatabaseManager($this->configuration);
 
     $excludeTypes = (!empty($options['exclude-types'])) ? explode(',', strtolower($options['exclude-types'])) : null;
 
     QubitSearch::getInstance()->populate($excludeTypes);
   }
+
+  protected function availableDocumentTypes()
```
Comment (MikeFE) [inline lib/task/search/arPopulateTask.class.php]: can be private, we're unlikely to inherit from this class and need protected

**Comment by MikeFE** [comment]
:+1: other than 2 nitpicks

File: lib/task/search/arPopulateTask.class.php (hunk 1 above)
Comment (mcantelon) [reply lib/task/search/arPopulateTask.class.php]: Unfortunately PHP's sort is weird and sorts in place, returning a boolean (success or failure).

File: lib/task/search/arPopulateTask.class.php (hunk 2 above)
Comment (mcantelon) [reply lib/task/search/arPopulateTask.class.php]: Ah true, will change that.

**Comment by mcantelon** [comment]
Merged.

**Pull request closed without merging**